newIDE/app/src/VariablesList/MultilineVariableEditorDialog.js

```diff
@@ -0,0 +1,47 @@
+// @flow
+import * as React from 'react';
+import { Trans } from '@lingui/macro';
+import Dialog, { DialogPrimaryButton } from '../UI/Dialog';
+import SemiControlledTextField from '../UI/SemiControlledTextField';
+
+type Props = {|
+  initialValue: string,
+  onClose: (newValue: string) => void,
+|};
+
+export const MultilineVariableEditorDialog = ({
+  initialValue,
+  onClose,
+}: Props) => {
+  const [value, setValue] = React.useState(initialValue);
+
+  return (
+    <Dialog
+      open
+      title={null}
+      flexColumnBody
+      actions={[
+        <DialogPrimaryButton
+          id="ok"
+          label={<Trans>Ok</Trans>}
+          primary
+          onClick={() => onClose(value)}
+        />,
+      ]}
+      maxWidth="md"
+      onRequestClose={() => onClose(value)}
```

Comment:
Shouldn't this cancel the changes?
Also, I see there is no cancel button, should you add one?

Reply:
I was wondering because this dialog is an "extension" of the field that you have in the editor, it's just a "big field that happens to show in a dialog", so the undo/redo should be done... in the variables editor itself :)